alws/crud/test.py

```diff
@@ -141,3 +147,66 @@ async def get_test_tasks_by_build_task(
             query = query.filter(models.TestTask.revision == revision)
         result = await db.execute(query)
         return result.scalars().all()
+
+
+async def get_test_logs(build_id: int, db: Session) -> dict:
+    """
+    Parses test logs and determine test format.
+    Returns dict of lists for each build's test with detailed status report.
+
+    Parameters
+    ----------
+    build_id : int
+        Id of the build, which tests info is interested
+    db: Session
+        db connection
+
+    Returns
+    -------
+    dict
+
+    """
+    async with db.begin():
+        test_task_query = select(models.TestTask).where(
+            models.TestTask.build_task_id == build_id)
+        result = await db.execute(test_task_query)
+        test_task = result.scalars().all()
+        repo_id = test_task[0].repository_id
+        repo_url_query = select(models.Repository.url).where(
+            models.Repository.id == repo_id)
+        result = await db.execute(repo_url_query)
+        repo_url = result.scalars().first()
+
+        test_task_ids = [test.id for test in test_task]
+
+        test_names_query = select(models.TestTaskArtifact).where(
+        models.TestTaskArtifact.test_task_id.in_(test_task_ids)
+        )
+        result = await db.execute(test_names_query)
+
+        test_artifacts = result.scalars().all()
+        test_names = defaultdict(list)
+        for artifact in test_artifacts:
+            if artifact.name.startswith('tests_'):
+                test_names[artifact.test_task_id].append(artifact.name)
+
+        test_results = []
+        for test_task_test_names in test_names.values():
```

Comment:
Are you sure you still need opened db session (with) or this code could be launce outside of with statement?